### ISSUE TITLE
Conflict rules: `newline-after-import` with `order`

### ISSUE THREAD

**Issue by Arttse** [benmosher/eslint-plugin-import#1811]
Maybe a bug, with `v2.20.2` works well.

```ts
// ESLint: There should be no empty line within import group(import/order)
import { ExecaReturnValue } from 'execa';

import execa = require('execa');
```

When remove 1 empty line:
```ts
// ESLint: Expected 1 empty line after import statement not followed by another import.(import/newline-after-import)
import { ExecaReturnValue } from 'execa';
import execa = require('execa');
```

---

<details>
<summary>Config rules</summary>

```js
'import/newline-after-import': ['error'],
'import/order': [
  'error',
  {
    groups: [
      ['builtin', 'external', 'internal'],
      'unknown',
      'parent',
      ['sibling', 'index'],
    ],
    'newlines-between': 'always',
  },
],
```
</details>

|                           |         |
|---------------------------|---------|
| OS                        | Linux   |
| node.js                   | 12.18.0 |
| eslint                    | 7.2.0   |
| eslint-plugin-import      | 2.21.1  |
| @typescript-eslint/parser | 3.1.0   |
| typescript                | 3.9.5   |

**Comment by ljharb** [comment]
This indeed looks like a bug in `newline-after-import` not understanding TypeScript's nonstandard `import =` syntax.

While I'd suggest enabling esModuleInterop and synthetic imports, and avoiding this syntax, this is still a bug we'll need to fix.
